features/frameworks/greentea-client/source/test_env.cpp

```diff
@@ -61,8 +61,10 @@ static void greentea_notify_version();
  * Rawserial object used to provide direct, raw serial communications
  * between the target and the host.
  */
-RawSerial greentea_serial(USBTX, USBRX); 
-
+RawSerial& greentea_serial() { 
+    static RawSerial serial(USBTX, USBRX);
```

Comment:
From what I understand, the static-initializer solution has several benefits over `LazyInstance`:
- Language-based solution, no delicate tip-toeing around the definition of a POD type
- Little concerns for regressions and one less class to support
- Can easily do non-trivial amount of work (and accept arguments)
- Is forward compatible with non-trivial changes to the construction of the object

If the only issue is the semantics of a reference over pointer, these functions could be easily changed to return a pointer.